src/main/java/net/pms/util/CustomImageReader.java

```diff
@@ -0,0 +1,147 @@
+package net.pms.util;
+
+import java.awt.image.BufferedImage;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Iterator;
+import java.util.Locale;
+import javax.imageio.IIOException;
+import javax.imageio.ImageIO;
+import javax.imageio.ImageReadParam;
+import javax.imageio.ImageReader;
+import javax.imageio.spi.IIORegistry;
+import javax.imageio.spi.ImageInputStreamSpi;
+import javax.imageio.stream.ImageInputStream;
+import net.pms.util.ImagesUtil.ImageFormat;
+
+/**
+ * This is a hack of a utility class created because whoever wrote {@link ImageIO}
+ * decided to make it {@code final}. The sole purpose of these methods is to
+ * return the image format of the source image along with the {@link BufferedImage}.
+ *
+ * There should be no reason to have to do this twice, and thus the result from
+ * the analysis done when reading the image is kept.
+ */
+public class CustomImageReader {
+
+    protected static final IIORegistry theRegistry = IIORegistry.getDefaultInstance();
+
+	// Not to be instantiated
+	private CustomImageReader() {
+	}
+
+	/**
+	 * A copy of {@link ImageIO#read(InputStream)} that calls {{@link #read(ImageInputStream)}
+	 * instead of {@link ImageIO#read(ImageInputStream)}.
+	 *
+	 * @see ImageIO#read(InputStream)
+	 */
+    public static ImageReaderResult read(InputStream input) throws IOException {
+        if (input == null) {
+            throw new IllegalArgumentException("input == null!");
+        }
+
+        ImageInputStream stream = createImageInputStream(input);
+        ImageReaderResult result = read(stream);
+        if (result.bufferedImage == null) {
```

Comment:
@Sami32 Thanks to your log, it seems like ```result``` can be null (even though I can't quite see how). This should fix it:
```java
        if (result == null || result.bufferedImage == null) {
```